ocs_ci/ocs/resources/storage_cluster.py

```
@@ -2844,16 +2844,56 @@ def check_custom_storageclass_presence(interface=None):
 
     sc_list = get_all_storageclass_names()
 
-    missing_sc = [value for value in sc_from_spec.values() if value not in sc_list]
+    # Check if underlying Ceph resources exist for managed resources
+    def check_underlying_resource_exists(storage_class_type):
+        """Check if the underlying Ceph resource exists."""
+        try:
+            namespace = config.ENV_DATA["cluster_namespace"]
+            resource_map = {
+                "cephFilesystems": "CephFilesystem",
+                "cephBlockPools": "CephBlockPool",
+                "cephObjectStores": "CephObjectStore",
+            }
+
+            if storage_class_type in resource_map:
+                kind = resource_map[storage_class_type]
+                resource_obj = OCP(kind=kind, namespace=namespace)
+                resources = resource_obj.get().get("items", [])
+                return len(resources) > 0
+            return True  # For nfs/encryption, assume resource exists
+        except Exception:
+            return True  # If we can't check, assume it exists
```

in what situation would that happen? Shouldn't we at least throw an error log?